src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2120,9 +2120,7 @@ ManageWorkerPool(WorkerPool *workerPool)
 
 		/* create a session for the connection */
 		session = FindOrCreateWorkerSession(workerPool, connection);
-
-		/* always poll the connection in the first round */
-		UpdateConnectionWaitFlags(session, WL_SOCKET_READABLE | WL_SOCKET_WRITEABLE);
+		ConnectionStateMachine(session);
```

Comment:
I merged this part of the PR.

Reply:
yes，i use the pg_auto_failover. adaptive/real-time/task-tracker test is success（select * from aa ; select * from aa where id = xx），